onnx/defs/math/defs.cc

```diff
@@ -83,29 +82,7 @@ will throw errors.
         "T",
         {"tensor(float16)", "tensor(float)", "tensor(double)"},
         "Constrain input and output types to float tensors.");
-    schema.TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
-      propagateElemTypeFromInputToOutput(ctx, 0, 0);
-      if (!hasNInputShapes(ctx, 1)) {
-        return;
-      }
-      propagateShapeFromInputToOutput(ctx, 0, 0);
-      const TensorShapeProto& input_shape =
-        ctx.getInputType(0)->tensor_type().shape();
-      int r = input_shape.dim_size();
-      if (r != 2) {
```

Comment:
This is incorrect as the input need not be 2D. This was a comment in the PR #2281  

Effectively once we know that the output shape is the same as the input shape, there is no point in any other validation. This is based on Rama's comment in #2132 - 

"I feel that validation and generating output shape are distinct sub-problems. We do the best we can for both. Even if we cannot do validation here, we can still generate information..."

So I just reverted the shape inference function altogether

Reply:
If we're doing the best we can we can still validate the axis so I think that should remain.

Reply:
Just to clarify the context and the point of my comment: there were cases where the inference function was "returning" while there was still useful steps it could perform (e.g., setting the output rank). My point was that we should DO it. This case is a bit different, since it is a "fail_shape_inference". If the rank is not required to be 2, it seems erroneous to enforce it, so dropping it seems the right thing. For the axis check: as Scott says, there seems nothing wrong with retaining it.

Reply:
Retained the validation logic. Thanks for clarifying. :)